sky/global_user_state.py

```diff
@@ -104,19 +131,50 @@ class StorageStatus(enum.Enum):
     READY = 'READY'
 
 
-def add_or_update_cluster(cluster_name: str,
-                          cluster_handle: 'backends.Backend.ResourceHandle',
-                          ready: bool,
-                          is_launch: bool = True):
+def add_or_update_cluster(
+    cluster_name: str,
+    cluster_handle: 'backends.Backend.ResourceHandle',
+    requested_resources: Optional[Set[Any]],
+    ready: bool,
+    is_launch: bool = True,
+):
     """Adds or updates cluster_name -> cluster_handle mapping."""
     # FIXME: launched_at will be changed when `sky launch -c` is called.
     handle = pickle.dumps(cluster_handle)
     cluster_launched_at = int(time.time()) if is_launch else None
     last_use = common_utils.get_pretty_entry_point() if is_launch else None
     status = ClusterStatus.UP if ready else ClusterStatus.INIT
+
+    cluster_hash = _get_cluster_hash(cluster_name) or str(uuid.uuid4())
```

Comment:
If our provisioning failover to another region, we will have multiple entries in the cluster history. It can become more inaccurate for multiple nodes, since, during failover, the multi-node cluster can be partially UP, then terminated and switched to another region. It might be fine for now, but we can add a comment:
```
TODO (sumanth): There will be multiple entries in the cluster history table when the cluster failover through multiple regions (one entry per region). It can be more inaccurate for the multi-node cluster as the failover can have the nodes partially UP.
```